examples/stats_models/regression.jl

```diff
@@ -0,0 +1,243 @@
+
+using SemanticModels
+using Random
+using LightGraphs
+using MetaGraphs
+using DataFrames
+using GraphPlot
+using DifferentialEquations
+using Plots
+using MultivariateStats
+using Statistics
+
+mutable struct DepVar{T}
+    value::T
+end
+
+mutable struct IndepVar{T}
+    value::T
+end
+
+# to allow transformations that feature numeric elements/vectors 
+mutable struct Other{T}
+    value::T
+end
+
+
+const Var = Union{DepVar, IndepVar}
+const RegComponents = Union{DepVar, IndepVar, Other}
+
+
+mutable struct OpElementTypes{a, b}
+    vec_a::a
+    vec_b::b
+end
+
+function var_type_check(a::Var, b::Other)
+    return isa(a, DepVar) ? DepVar : IndepVar
+end
+    
+function var_type_check(a::Other, b::Var)
+    return isa(b, DepVar) ? DepVar : IndepVar
+end
+
+function var_type_check(a::Var, b::Var)   
+    return isa(a, DepVar) || isa(b, DepVar) ? DepVar : IndepVar
+
+end
+
+function var_type_check(a::Other, b::Other)   
+    return Other
+end
+
+
+binary_ops = [+,-,*,/]
+
+# example loops to get element-level truth tables 
+dv_num = 10
+iv_num = 3
+other_num = 7
+
+
+for op in binary_ops
+    for a in (DepVar(dv_num), IndepVar(iv_num), Other(other_num))
+        for b in (DepVar(dv_num), IndepVar(iv_num), Other(other_num))
+            output_check = var_type_check(a,b)
+            result_type = output_check(op(a.value, b.value))
+            print("Op: $op, a: $a, b: $b, result_type: $result_type \n")
+        end
+    end
+end
+
+function transform_col_vector(op, vec_a::Array, vec_b::Array)
+    
+    if length(vec_a) == length(vec_b)
+
+        output_types = map(var_type_check, [x for x in vec_a], [x for x in vec_b]) 
```

Comment:
If you just do `broadcast(op, a, b)` it should work without the `[x.value for x in vec_a]` because broadcast uses the element wise functions.

Reply:
The key to making this work is to define the new method for `+(a::DepVar, b::IndepVar) = DepVar(a.value + b.value)`. I think you have to use `eval` above in order to make that method exist.

Reply:
done